nixos/modules/services/networking/headscale.nix

```
@@ -0,0 +1,486 @@
+in
+{
+  options = {
+    services.headscale = {
+      enable = mkEnableOption ''
+        Whether to enable the headscale coordination server for Tailscale.
+      '';
+
+      package = mkOption {
+        default = pkgs.headscale;
+        defaultText = literalExpression "pkgs.headscale";
+        type = types.package;
+        description = "headscale package to use.";
+      };
+
+      user = mkOption {
+        default = "headscale";
+        type = types.str;
+        description = ''
+          User the headscale server should execute under.
+        '';
+      };
+
+      group = mkOption {
+        default = "headscale";
+        type = types.str;
+        description = ''
+          If the default user "headscale" is configured then this is the primary
+          group of that user.
+        '';
+      };
+
+      dataDir = mkOption {
+        default = "/var/lib/headscale";
+        type = types.path;
+        description = ''
+          Directory to store data for the headscale server.
+        '';
+      };
+
+      runDir = mkOption {
+        default = "/run/headscale";
+        type = types.path;
+        description = ''
+          Directory for runtime socket.
+        '';
+      };
+
+      serverUrl = mkOption {
+        type = types.str;
+        default = "http://127.0.0.1:8080";
+        description = ''
+          The url clients will connect to.
+        '';
+        example = "https://myheadscale.example.com:443";
+      };
+
+      address = mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        description = ''
+          Listening address of headscale
+        '';
+        example = "0.0.0.0";
+      };
+
+      port = mkOption {
+        type = types.int;
+        default = 8080;
+        description = ''
+          Listening port of headscale
+        '';
+        example = "443";
+      };
+
+      privateKeyFile = mkOption {
+        default = "";
```

is this a mandatory option? if so it's better to not give a default and get an error out of the module system if it's not set